src/core/shouldUpdateReactComponent.js

```diff
@@ -63,19 +63,23 @@ function shouldUpdateReactComponent(prevElement, nextElement) {
             if (typeof nextElement.type !== 'string' ||
                 nextElement.type === 'input' ||
                 nextElement.type === 'textarea') {
-              warning(
-                false,
-                '<%s /> is being rendered by both %s and %s using the same key ' +
-                '(%s) in the same place. Currently, this means that they ' +
-                'don\'t preserve state. This behavior should be very rare ' +
-                'so we\'re considering deprecating it. Please contact the ' +
-                'React team and explain your use case so that we can take that ' +
-                'into consideration.',
-                nextDisplayName || 'Unknown Component',
-                prevName || '[Unknown]',
-                nextName || '[Unknown]',
-                prevElement.key
-              );
+              if (nextElement._owner != null &&
+                  nextElement._owner._isOwnerNecessray === undefined) {
```

Comment:
Combine with above conditional? The only content in this block is the `if`. Do we foresee anything else coming in here?